datasets/humicroedit/humicroedit.py

```diff
@@ -0,0 +1,170 @@
+import os
+
+import datasets
+
+
+_CITATION = """\
+@article{hossain2019president,
+  title={" President Vows to Cut< Taxes> Hair": Dataset and Analysis of Creative Text Editing for Humorous Headlines},
+  author={Hossain, Nabil and Krumm, John and Gamon, Michael},
+  journal={arXiv preprint arXiv:1906.00274},
+  year={2019}
+}
+"""
+
+_DESCRIPTION = """\
+This new dataset is designed to assess the funniness of edited news headlines.
+"""
+
+_HOMEPAGE = "https://www.cs.rochester.edu/u/nhossain/humicroedit.html"
+_LICENSE = ""
+
+_URL = "https://cs.rochester.edu/u/nhossain/semeval-2020-task-7-dataset.zip"
+
+
+class Humicroedit(datasets.GeneratorBasedBuilder):
+    """This is humorous headline dataset called Humicroedit introduced in the Task-7 of SemEval 2020."""
+
+    VERSION = datasets.Version("1.1.0")
+
+    # This is an example of a dataset with multiple configurations.
+    # If you don't want/need to define several sub-sets in your dataset,
+    # just remove the BUILDER_CONFIG_CLASS and the BUILDER_CONFIGS attributes.
+
+    # If you need to make complex sub-parts in the datasets with configurable options
+    # You can create your own builder configuration class to store attribute, inheriting from datasets.BuilderConfig
+    # BUILDER_CONFIG_CLASS = MyBuilderConfig
+
+    # You will be able to load one or the other configurations in the following list with
+    # data = datasets.load_dataset('my_dataset', 'first_domain')
+    # data = datasets.load_dataset('my_dataset', 'second_domain')
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="subtask-1", description="This part of the dataset covers the data for subtask-1"),
+        datasets.BuilderConfig(name="subtask-2", description="This part of the dataset covers the data for subtask-2"),
+    ]
+
+    def _info(self):
+        if self.config.name == "subtask-1":
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "original": datasets.Value("string"),
+                    "edit": datasets.Value("string"),
+                    "grades": datasets.Value("string"),
+                    "meanGrade": datasets.Value("float"),
+                    # These are the features of your dataset like images, labels ...
+                }
+            )
+        else:
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "original1": datasets.Value("string"),
+                    "edit1": datasets.Value("string"),
+                    "grades1": datasets.Value("string"),
+                    "meanGrade1": datasets.Value("float"),
+                    "original2": datasets.Value("string"),
+                    "edit2": datasets.Value("string"),
+                    "grades2": datasets.Value("string"),
+                    "meanGrade2": datasets.Value("float"),
+                    "label": datasets.ClassLabel(names=[0, 1, 2]),
```

Comment:
Here we expect the names to be meaningful names as strings.

For example you can just change it to
```suggestion
                    "label": datasets.ClassLabel(names=["equal", "sentence1", "sentence2"]),
```

after changing this can you regenerate the dataset_infos.json file ?

Reply:
@lhoestq 
I made the change suggested by you. But dummy data generation and generation of dataset_infos.json fails. It gives the Key error as shown below:
```Traceback (most recent call last):
  File "datasets-cli", line 36, in <module>
    service.run()
  File "/home/saradhix/datasets/src/datasets/commands/dummy_data.py", line 321, in run
    keep_uncompressed=self._keep_uncompressed,
  File "/home/saradhix/datasets/src/datasets/commands/dummy_data.py", line 359, in _autogenerate_dummy_data
    dataset_builder._prepare_split(split_generator)
  File "/home/saradhix/datasets/src/datasets/builder.py", line 905, in _prepare_split
    example = self.info.features.encode_example(record)
  File "/home/saradhix/datasets/src/datasets/features.py", line 799, in encode_example
    return encode_nested_example(self, example)
  File "/home/saradhix/datasets/src/datasets/features.py", line 710, in encode_nested_example
    (k, encode_nested_example(sub_schema, sub_obj)) for k, (sub_schema, sub_obj) in utils.zip_dict(schema, obj)
  File "/home/saradhix/datasets/src/datasets/features.py", line 710, in <genexpr>
    (k, encode_nested_example(sub_schema, sub_obj)) for k, (sub_schema, sub_obj) in utils.zip_dict(schema, obj)
  File "/home/saradhix/datasets/src/datasets/features.py", line 737, in encode_nested_example
    return schema.encode_example(obj)
  File "/home/saradhix/datasets/src/datasets/features.py", line 522, in encode_example
    example_data = self.str2int(example_data)
  File "/home/saradhix/datasets/src/datasets/features.py", line 481, in str2int
    output.append(self._str2int[str(value)])
KeyError: '1'
```
I tried to fix by making changes in the generate_examples method by returning the label as one of the label names defined. I made this change in generate_examples for subtask2 as shown below

```
label_names=["equal", "sentence1", "sentence2"]
...
"label": label_names[int(label)],
```
With this change, the error is gone, but I am not sure if this is the right fix. Please confirm if I can proceed with this fix.

Reply:
Yes this is the right fix !

Reply:
Pushed the changes.